src/xrpld/app/tx/detail/Payment.cpp

```diff
@@ -261,18 +261,31 @@ Payment::checkPermission(ReadView const& view, STTx const& tx)
     if (checkTxPermission(sle, tx) == tesSUCCESS)
         return tesSUCCESS;
 
+    auto const& dstAmount = tx.getFieldAmount(sfAmount);
+
+    // If fixDelegateV1_1 is not enabled, PaymentMint/PaymentBurn is not
+    // supported for MPT. In this case, only transaction-level permission
+    // (specifically Payment) is allowed for MPT.
+    if (!view.rules().enabled(fixDelegateV1_1) && dstAmount.holds<MPTIssue>())
+        return tecNO_DELEGATE_PERMISSION;
```

Comment:
does this return statement change the order of execution in any way? what would be returned if this code is omitted?

If the results differ, adding this return would be a breaking change.

Reply:
I just tried commenting this block out. It seems like one of the tests would result in tecNO_DELEGATE_PERMISSION instead of tesSuccess. In this case, we can't add this code here